### ISSUE TITLE
No way to add alt text to images

### ISSUE THREAD

**Issue by myasonik** [mattermost/mattermost-server#19193]
#### Summary
When uploading an image, there is no way for me to add alt text to that image

#### Steps to reproduce
Mattermost Version: 5.38.0

#### Expected behavior
A user should be able to upload an image with alt text

#### Observed behavior (that appears unintentional) 
No way to associate alt text with an image

#### Possible fixes
Need a bit more user flow while uploading an image. Allowing an admin to make alt text a requirement would also be nice

**Comment by amyblais** [comment]
@myasonik What do you see when following the instructions here https://docs.mattermost.com/messaging/formatting-text.html#in-line-images?

**Comment by myasonik** [comment]
Oh, that does work! Great!

The docs for that were difficult enough to find that my team couldn't. Some more accessibility-related keywords in there might help? Not sure really, discoverability is a harder problem.

Is there a way to add alt text when uploading an image?

**Comment by amyblais** [comment]
> The docs for that were difficult enough to find that my team couldn't. Some more accessibility-related keywords in there might help? Not sure really, discoverability is a harder problem.

Thanks for the feedback, feel free to send this as feedback either via the feedback form at the bottom of the doc (see attached screenshot) or by opening a Github issue at https://github.com/mattermost/docs/issues.

> Is there a way to add alt text when uploading an image?

@hmhealey Would you know?

**Comment by hmhealey** [comment]
No, I don't think there's currently a way to do that for file attachments.

A fairly unintrusive thing that we could do would be to use the file name as alt text, but that's only helpful sometimes since a lot of the time it'd just be "screenshot at 1:15pm.jpg" or complete gibberish. We could add an extra popup to attaching a file that would prompt the user for a text description, but that could be a bit annoying to do every time. Maybe we could add it as a menu option that appears on the file attachment in the draft area? Anyway, those are just a few ideas after brainstorming for a few minutes

**Comment by amyblais** [comment]
Feel free to share any improvement ideas via https://mattermost.uservoice.com/forums/306457-general.

**Comment by myasonik** [comment]
Looks like uservoice is for new feature suggestions, not for existing bugs. Is there another place I should leave a bug report or can this be reopened?

**Comment by amyblais** [comment]
Sorry, I originally thought this as an improvement idea - I've opened a bug ticket now https://mattermost.atlassian.net/browse/MM-41686.

**Comment by myasonik** [comment]
@hmhealey I can't respond to [your comment in Jira](https://mattermost.atlassian.net/browse/MM-41686?focusedCommentId=122521) so I hope you don't mind if I respond here. 

I really disagree on calling this issue a feature request. This is foundational accessibility that's native to the web. The bug is that Mattermost is not exposing any mechanism to use it. 

[Mattermost's VPAT](https://docs.mattermost.com/about/vpat.html#section-1194-22-web-based-intranet-and-internet-information-and-applications-detail) also claims that this was 80-90% supported throughout the app whenever it was audited and that y'all were planning to resolve it in 2018. I realize that y'all probably only considered images that y'all own at the time but as an authoring tool (in that people post random content to it), I need to be able to post accessible content otherwise what was the point of an accessible product?

This also flies in the face of [last years' blog post on accessibility](https://mattermost.com/blog/mattermost-accessibility-for-the-visually-impaired/) that starts out with:
> Mattermost is dedicated to providing effective solutions that can be utilized by as many people as possible. Accessibility is central to our core belief that chat should be simple, intuitive, and functional.

And ends with:
> Any technical issue in a current or future product release that would prevent compliance with accessibility ratings stated in product documentation would be considered a product defect and Mattermost would welcome the public filing of an issue report against the defect so that it may be resolved.

**Comment by hmhealey** [comment]
Sorry, I didn't mean to be dismissive about this issue. This is definitely something we should add, but I was referring to bug vs feature in a different sense where a bug is something that isn't working as designed where a dev would know what to fix vs an improvement which might need new designs or UI elements. That was more for our internal routing and to get the right eyes on this issue since we need a PM or designer to look at this first rather than having it get sent directly to a dev.

As for that assessment, I agree that it's likely referring to the built-in images since I remember a bunch of alt tags or aria-labels being added to things like profile images and icons back then. I think these image attachments might technically satisfy that requirement since they have a "file attachment" aria-label, but that obviously isn't helpful to an actual user.

Anyway, I'm going to also bump this on our community server to make sure the right people take a look at this. It does need some new UI elements to be able to add this, but it seems relatively easy to implement once we get those designs sorted out, and it'll definitely be a big improvement once that's in

**Comment by myasonik** [comment]
Sorry about jumping to conclusions about the phrasing. Thank you for clarifying some of the internal mechanicians and for continuing to dig into the issue!

**Comment by hmhealey** [comment]
No worries at all. I'll admit I definitely was a bit dismissive about this at first as well, so I'm glad I'm understanding the issue better from an accessibility standpoint now 🙂

**Comment by lieut-data** [comment]
Thanks for reporting this issue and for the community discussion around it. After multiple years of inactivity, we're closing this to help keep our issue tracker focused on active work. If this is still relevant, please feel free to open a new issue with updated context and we'll be happy to take another look.
